core/src/main/java/com/scalar/db/common/CheckedDistributedStorageAdmin.java

```diff
@@ -0,0 +1,277 @@
+package com.scalar.db.common;
+
+import com.scalar.db.api.DistributedStorageAdmin;
+import com.scalar.db.api.TableMetadata;
+import com.scalar.db.exception.storage.ExecutionException;
+import com.scalar.db.io.DataType;
+import com.scalar.db.util.ScalarDbUtils;
+import java.util.Map;
+import java.util.Set;
+import javax.annotation.Nullable;
+
+public class CheckedDistributedStorageAdmin implements DistributedStorageAdmin {
+
+  private final DistributedStorageAdmin admin;
+
+  /**
+   * Whether to check if the namespace exists or not. Set false when the storage does not support
+   * namespaces.
+   */
+  private final boolean checkNamespace;
+
+  public CheckedDistributedStorageAdmin(DistributedStorageAdmin admin) {
+    this(admin, true);
+  }
+
+  public CheckedDistributedStorageAdmin(DistributedStorageAdmin admin, boolean checkNamespace) {
+    this.admin = admin;
+    this.checkNamespace = checkNamespace;
+  }
+
+  @Override
+  public void createNamespace(String namespace, Map<String, String> options)
+      throws ExecutionException {
+    if (checkNamespace && namespaceExists(namespace)) {
+      throw new IllegalArgumentException("Namespace already exists: " + namespace);
+    }
+
+    try {
+      admin.createNamespace(namespace, options);
+    } catch (ExecutionException e) {
+      throw new ExecutionException("Creating the namespace failed: " + namespace, e);
+    }
+  }
+
+  @Override
+  public void createTable(
+      String namespace, String table, TableMetadata metadata, Map<String, String> options)
+      throws ExecutionException {
+    if (checkNamespace && !namespaceExists(namespace)) {
+      throw new IllegalArgumentException("Namespace does not exist: " + namespace);
+    }
+    if (tableExists(namespace, table)) {
+      throw new IllegalArgumentException("Table already exists: " + namespace + "." + table);
```

Comment:
```suggestion
      throw new IllegalArgumentException("Table already exists: " + ScalarDbUtils.getFullTableName(namespace, table));
```
Should the utility function be used all the time?

Reply:
Fixed in fcb233812ead7d8c6a68a422a7467f70ba10c733. Thanks.